Dockerfile

```
@@ -0,0 +1,9 @@
+FROM node:12-alpine
+WORKDIR /app
+COPY ./server.js ./package*.json ./webpack.config.js ./tsconfig.json ./
+COPY ./src ./src
+COPY ./public ./public
+RUN npm install && npm run build
+RUN rm -rf ./src && rm -rf ./public
+EXPOSE 3000
+CMD node server.js 
```

Файл можно сделать короче и даже выиграть в размере образа (255мб против 274мб)
```suggestion
FROM node:12-alpine
WORKDIR /app
COPY . .
RUN npm install && npm run build
CMD node server.js 
```

Если хочется совсем экстрима, то можно сделать следующее:

1. Убрать сервер в отдельный файл и сделать входной точкой приложения "голый" индекс
2. Написать для него отдельный вебпак-конфиг (чтобы скомпилить его и не тянуть `node_modules`)
3. Применить multi-stage build
4. На первом шаге поставить зависимости и сбилдить проект
5. На втором шаге забрать в итоговый образ только то, что нам нужно

```
FROM node:12-alpine as build
WORKDIR /app
COPY . .
RUN npm install && npm run build
RUN npx webpack --config webpack.server.config.js // структуру делал на коленке, в таких случаях конфиги вебпака кладут в отдельную папку

FROM node:12-alpine
COPY --from=build /app/dist /dist
COPY --from=build /app/index.js /
CMD node index.js
```

Таким образом вес образа будет в 3 раза мнеьше изначального. Я запушу тестовую ветку, где будет всё необходимое для такого подхода, посмотри потом, если интересно. На аппрув не повлияет, это если хочется запариться)

---

С учетом того, что сервер отдельно не компилится, вынесла сервер вместе с его модулями в отдельную папку. Получилось 95 мб против 275. Посмотри, пожалуйста, как тебе такой подход